flask_restful/__init__.py

```diff
@@ -287,33 +297,86 @@ def handle_error(self, e):
             else:
                 raise e
 
-        headers = Headers()
-        if isinstance(e, HTTPException):
-            code = e.code
-            default_data = {
-                'message': getattr(e, 'description', http_status_message(code))
-            }
-            headers = e.get_response().headers
+        for typecheck, handler in self.errorhandlers:
+            if isinstance(e, typecheck):
+                try:
+                    # unpack(None) -> (None, _, _)
+                    data, code, headers = unpack(handler(e))
+                    if data:
+                        break
+                except:
```

Comment:
Never use `except` without an exception class as this will also catch `SystemExit`, `KeyboardInterrupt`, and `GeneratorExit`. It's better to just catch `Exception` instead if you want a generic "catch-all" handler.